core/src/test/java/org/apache/calcite/test/JdbcAdapterTest.java

```diff
@@ -135,6 +135,14 @@ class JdbcAdapterTest {
             + "WHERE \"product_id\" = 1");
   }
 
+  @Test void testFilterUnionIncludingWithPlan() {
```

Comment:
Since the exception is thrown in the validating phase, it's better to add the test to `SqlValidatorTest`

Reply:
I added more checks to make this test fit into this file.